### PR TITLE
A variety of clarity changes to the your-first-game tutorial

### PR DESCRIPTION
**Pull request by casey-mc**
Tried to clear up some points of confusion I'd had when I went through the tutorial. Also added the rationale behind using certain techniques, like `_process` or `CanvasLayer`.

### CODE REVIEW

File: learning/step_by_step/your_first_game.rst [hunk 1]
```diff
@@ -316,6 +323,8 @@ create any number of independent mobs in the game.
 Node Setup
 ~~~~~~~~~~
 
+Click Scene -> New Scene and we'll create the Mob!
```
Comment (cbscribe) [inline learning/step_by_step/your_first_game.rst]: I don't like the "!" here.  How about:

Click Scene -> New Scene to begin creating the Mob.

File: learning/step_by_step/your_first_game.rst [hunk 2]
```diff
@@ -325,9 +334,9 @@ The Mob scene will use the following nodes:
    -  :ref:`VisibilityNotifier2D <class_VisibilityNotifier2D>` (named ``Visibility``)
 
 In the :ref:`RigidBody2D <class_RigidBody2D>` properties, set ``Gravity Scale`` to ``0`` (so
-that the mob will not fall downward). In addition, under
-``PhysicsBody2D``, click the ``Mask`` property and uncheck the first
-box. This will ensure that the mobs do not collide with each other.
+that the mob will not fall downward). In addition, under the
+``PhysicsBody2D`` section of the Inspector, click the ``Mask`` property and
```
Comment (cbscribe) [inline learning/step_by_step/your_first_game.rst]: "in the Inspector"

File: learning/step_by_step/your_first_game.rst [hunk 3]
```diff
@@ -670,6 +687,12 @@ Connect the ``timout()`` signal of ``MessageTimer`` and the
 Connecting HUD to Main
 ~~~~~~~~~~~~~~~~~~~~~~
 
+Now that we're done creating the ``HUD`` scene, save it and go back to ``Main``.
+Instance the ``HUD`` scene in ``Main`` like you did the ``Player`` scene, and place it at the
+bottom of tree. The full tree should look like this, so make sure you didn't miss anything!
```
Comment (cbscribe) [inline learning/step_by_step/your_first_game.rst]: change the ! to a :

**Comment by casey-mc** [comment]
Great suggestions! I sent a new commit to address them. Thanks for the help and let me know if there's anything else you feel should be changed.

**Comment by NathanLovato** [comment]
Thanks for the contribution. Seems all issues were solved, let's merge.

**Pull request merged**